zeebe/atomix/cluster/src/test/java/io/atomix/raft/ControllableRaftContexts.java

```diff
@@ -213,6 +212,16 @@ public RaftContext createRaftContext(
     return raft;
   }
 
+  private RaftContext addRaftServer(final MemberId memberId, final RaftContext raftContext) {
+    final var currRaftContext = raftServers.get(memberId);
+    if (currRaftContext != null) {
+      currRaftContext.close();
+    }
+
+    raftServers.put(memberId, raftContext);
```

Comment:
`MemberId` is created by `nodeId` which is just created from the sequence `1...nodeCount`.  nodeCount is defined once, when the `ControllableRaftContexts.java` get constructed (final class variable). `raftServers.put(memberId, raftContext);` get called over and over, during the tests, with the same key - memberId.